### PR TITLE
SRCH-94

### PR DESCRIPTION
**Pull request by juliacollins**
opensearch-browser provides two options for communication: XMLHttpRequest and fetch. For fetch support, it's unfortunately depending on an older version of isomorphic-fetch, which in turn depends on a version of node-fetch with critical security vulnerabilities. I've configured search-interface to use the XMLHttpRequest option, which I _think_ is safe, but ultimately the package needs to be updated. I'll submit a pull request to opensearch-browser to update to the latest version of isomorphic-fetch. My question is: Does using XMLHttpRequest actually avoid the security issue?

### CODE REVIEW

File: audit-resolve.json [hunk 1]
```diff
@@ -4,8 +4,16 @@
...
       "madeAt": 1626802771020,
       "expiresAt": 1640995200000
+    },
+    "1091239|opensearch-browser>isomorphic-fetch>node-fetch": {
+      "decision": "ignore",
+      "madeAt": 1692723196646
+    },
+    "1091791|opensearch-browser>isomorphic-fetch>node-fetch": {
+      "decision": "ignore",
+      "madeAt": 1692723202247
```
Comment (juliacollins) [inline audit-resolve.json:14]: These changes allow the CI workflow to ignore the security vulnerabilities being report for opensearch-browser/isomorphic-fetch/node-fetch.

File: audit-resolve.json [hunk 2]
```diff
@@ -4,8 +4,16 @@
...
     }
   },
   "rules": {},
   "version": 1
-}
+}
```
Comment (scott-lewis-nsidc) [inline audit-resolve.json:19]: Should there be a newline to end this file?  (Maybe there is, but the icon at the end makes me think that there isn't one?  Not sure what that means).

**Review by scott-lewis-nsidc — APPROVED** [review]
Looks good from what I can tell.  Had one comment about a missing newline, but I'll let you make the call on that one.

File: audit-resolve.json (hunk 2 above)
Comment (juliacollins) [reply audit-resolve.json:19]: Good question. I'll take a look.

**Comment by juliacollins** [comment]
@scott-lewis-nsidc I've made a few more updates, whenever you have a chance to take a look. The weird use of a branch for installing `OpenSearchlight` is now replaced with the weird use of a branch in a forked repository for `opensearch` :-). I did submit a PR for the `opensearch` package changes, and heard back from the repo owner, so I think they will review and eventually merge the PR. Since the timing of that activity is unknown, I'll add another story to revisit this configuration once again down the line. I neglected to create a new branch with my latest changes, but I believe the commits on August 25 are all of the changes made after your initial review.

**Comment by scott-lewis-nsidc** [comment]
> @scott-lewis-nsidc I've made a few more updates, whenever you have a chance to take a look. The weird use of a branch for installing `OpenSearchlight` is now replaced with the weird use of a branch in a forked repository for `opensearch` :-). I did submit a PR for the `opensearch` package changes, and heard back from the repo owner, so I think they will review and eventually merge the PR. Since the timing of that activity is unknown, I'll add another story to revisit this configuration once again down the line. I neglected to create a new branch with my latest changes, but I believe the commits on August 25 are all of the changes made after your initial review.

All looks good to me!

**Comment by juliacollins** [comment]
@scott-lewis-nsidc One last commit to point to the updated tag on the opensearch-browser package, if you care to take a look.

**Review by scott-lewis-nsidc — APPROVED** [review]
Looks good!

**Pull request merged**